sdk/core/Azure.Core.TestFramework/src/Instrumentation/DiagnosticScopeValidatingInterceptor.cs

```diff
@@ -202,6 +203,12 @@ internal static async ValueTask<T> ValidateDiagnosticScope<T>(Func<ValueTask<(T
                             throw new InvalidOperationException($"All diagnostic scopes should have 'az.namespace' attribute, make sure the assembly containing **ClientOptions type is marked with the AzureResourceProviderNamespace attribute specifying the appropriate provider. This attribute should be included in AssemblyInfo, and can be included by pulling in AzureResourceProviderNamespaceAttribute.cs using the AzureCoreSharedSources alias.");
                         }
 
+                        Activity parent = e.Activity.Parent;
+                        if (parent != null && !parent.Tags.Any(tag => tag.Key == "kind" && (tag.Value == "client" || tag.Value == "internal")))
+                        {
+                            throw new InvalidOperationException($"Client diagnostic scopes should not be nested except for backward compatibility reasons. Use ClientDiagnostics constructor overload to suppress nested scopes.");
```

Comment:
The guidance in the exception message confused me a bit. I think we want people to set SuppressScopes to true if they are using DPG, but this may not actually be the cause of the issue. This exception could occur in an existing HLC or even in a grow-up DPG, where the root cause isn't that they need to SuppressScopes - but that there may be a usage issue where they have nested scopes in the convenience layer.

Reply:
Up to you, but it may be easier to just flip to a project reference where needed in this PR, and then we can flip everything back after the Core release.